### PR TITLE
Declare missing connection attribute

### LINKED ISSUE #14885: Declare connection for addons where missing
In order for the new connection declaration to be really useful, for example for filtering when installing addons or on the documentation website, we should declare the remaining addons.

So far many contributors already declared it for addons known to them, but still a lot are missing. This issue is created for the purpose of tracking this progress.

The documentation is here, although not yet regenerated after merging openhab/openhab-docs#2069: https://next.openhab.org/docs/developer/addons/addon.html#xml-structure-for-add-on-definitions

## Current status
_Updated 30.04.2023_

```shell
[openhab-addons]$ grep -R "<connection>" --include addon.xml | wc -l
368
[openhab-addons]$ grep -RL "<connection>" --include addon.xml | wc -l
27
```

### The list

```shell
grep -RL "<connection>" --include addon.xml | sed 's/^bundles\/org\.openhab\.\(.*\)\/src\(.*\)/- \1/' | sort
```

- io.homekit
- io.hueemulation
- io.imperihome
- io.metrics
- io.neeo
- io.openhabcloud
- persistence.dynamodb
- persistence.influxdb
- persistence.jdbc
- persistence.jpa
- persistence.mapdb
- persistence.mongodb
- persistence.rrd4j
- transform.bin2json
- transform.exec
- transform.jinja
- transform.jsonpath
- transform.map
- transform.regex
- transform.rollershutterposition
- transform.scale
- transform.xpath
- transform.xslt
- voice.actiontemplatehli
- voice.porcupineks
- voice.rustpotterks
- voice.voskstt

### PR DESCRIPTION
**Pull request by holgerfriedrich**
For those addons the new attribute "connection" has not yet been declared.
Fixes #14885.

See https://next.openhab.org/docs/developer/addons/addon.html#xml-structure-for-add-on-definitions
for further information on the attribute.

This patch does not include io.imperihome, as I have opened a PR to remove it from the repo.

### CODE REVIEW

File: bundles/org.openhab.voice.voskstt/src/main/resources/OH-INF/addon/addon.xml [hunk 1]
```diff
@@ -6,6 +6,7 @@
 	<type>voice</type>
 	<name>Vosk Speech-to-Text</name>
 	<description>osk STT Service uses vosk-api to perform offline speech-to-text in openHAB.</description>
```
Comment (wborn) [inline bundles/org.openhab.voice.voskstt/src/main/resources/OH-INF/addon/addon.xml:8]:
```suggestion
	<description>Vosk STT Service uses vosk-api to perform offline speech-to-text in openHAB.</description>
```

File: bundles/org.openhab.transform.exec/src/main/resources/OH-INF/addon/addon.xml [hunk 2]
```diff
@@ -6,5 +6,6 @@
 	<type>transformation</type>
 	<name>Exec transformation</name>
 	<description>Transforms an input string with an external program.</description>
+	<connection>local</connection>
```
Comment (jlaur) [inline bundles/org.openhab.transform.exec/src/main/resources/OH-INF/addon/addon.xml:9]: Or is it `none`? I assume the external program is being executed on the machine running openHAB, not through the network?

File: bundles/org.openhab.voice.voskstt/src/main/resources/OH-INF/addon/addon.xml (hunk 1 above)
Comment (holgerfriedrich) [reply bundles/org.openhab.voice.voskstt/src/main/resources/OH-INF/addon/addon.xml:8]: Hm, I thought I had fixed this?!

File: bundles/org.openhab.transform.exec/src/main/resources/OH-INF/addon/addon.xml (hunk 2 above)
Comment (holgerfriedrich) [reply bundles/org.openhab.transform.exec/src/main/resources/OH-INF/addon/addon.xml:9]: I don't have a strong opinion here. For me, transforms typically are handled within openHAB -> none. The exec transform is a bit different, interacting with the system outside of openHAB -> so I thought local would be fine.

For the TTS addons we could have the same discussion - but most of the other TTS systems were tagged as local as well, so I have put local as well (although none would have been an option as well for some of them).

The connection attribute is a bit dependent on the view point of the author.

File: bundles/org.openhab.voice.voskstt/src/main/resources/OH-INF/addon/addon.xml [hunk 3]
```diff
@@ -5,7 +5,8 @@
 
 	<type>voice</type>
 	<name>Vosk Speech-to-Text</name>
-	<description>osk STT Service uses vosk-api to perform offline speech-to-text in openHAB.</description>
+	<description>Vosk STT Service uses vosk-api to perform offline speech-to-text in openHAB.</description>
+	<connection>local</connection>
```
Comment (GiviMAD) [inline bundles/org.openhab.voice.voskstt/src/main/resources/OH-INF/addon/addon.xml:9]: I think that connection can be none, Vosk library is included in the bundle and executed locally by the add-on.

File: bundles/org.openhab.voice.rustpotterks/src/main/resources/OH-INF/addon/addon.xml [hunk 4]
```diff
@@ -7,6 +7,7 @@
 	<name>Rustpotter Keyword Spotter</name>
 	<description>This voice service allows you to use the open source library Rustpotter as your keyword spotter in
 		openHAB.</description>
+	<connection>local</connection>
```
Comment (GiviMAD) [inline bundles/org.openhab.voice.rustpotterks/src/main/resources/OH-INF/addon/addon.xml:10]: Same than for Vosk, I think that connection can be none, Rustpotter library is included in the bundle and executed locally by the add-on.

**Review by GiviMAD — CHANGES_REQUESTED** [review]

**Comment by holgerfriedrich** [comment]
@jlaur @GiviMAD @wborn I followed the recommendation and set more connection types to none as requested.
For consistency of the voice addons, I think mactts, marytts, and picotts should also be tagged as none.
I have added this to the PR as well.

**Review by GiviMAD — APPROVED** [review]

**Review by jlaur — APPROVED** [review]
Thanks!

**Pull request merged**